opyn/opyn/wallet.py

```diff
@@ -133,9 +79,47 @@ def sign_bid_data(self, domain: Domain, message_to_sign: MessageToSign) -> BidDa
         if message_to_sign.signerAddress != self.public_key:
             raise ValueError("Signer wallet address mismatch")
 
-        signature = self._sign_type_data_v4(domain, asdict(message_to_sign), MESSAGE_TYPES)
-        print('signature', signature)
-        
+        data = {
+            "types": {
+                "EIP712Domain": [
+                    {"name": "name", "type": "string"},
+                    {"name": "version", "type": "string"},
+                    {"name": "chainId", "type": "uint256"},
+                    {"name": "verifyingContract", "type": "address"},
+                ],
+                "RFQ": [
+                    {"name": "offerId", "type": "uint256"},
+                    {"name": "bidId", "type": "uint256"},
+                    {"name": "signerAddress", "type": "address"},
+                    {"name": "bidderAddress", "type": "address"},
+                    {"name": "bidToken", "type": "address"},
+                    {"name": "offerToken", "type": "address"},
+                    {"name": "bidAmount", "type": "uint256"},
+                    {"name": "sellAmount", "type": "uint256"},
+                    {"name": "nonce", "type": "uint256"},
+                ],
+            },
+            "domain": {
+                "name": domain.name,
+                "version": domain.version,
+                "chainId": domain.chainId,
+                "verifyingContract": domain.verifyingContract,
+            },
+            "primaryType": "RFQ",
+            "message": {
+                "offerId": message_to_sign.offerId,
+                "bidId": message_to_sign.bidId,
+                "signerAddress": message_to_sign.signerAddress,
+                "bidderAddress": message_to_sign.bidderAddress,
+                "bidToken": message_to_sign.bidToken,
+                "offerToken": message_to_sign.offerToken,
+                "bidAmount": message_to_sign.bidAmount,
+                "sellAmount": message_to_sign.sellAmount,
+                "nonce": message_to_sign.nonce,
+            },
+        }
```

Comment:
To have a more compact configuration, both message and domain could be replaced with dataclasses.asdict and if we also move back the types to a global configuration we could reduce this to something like:

```
data2 = {
    "types": MESSAGE_TYPES,
    "domain": asdict(domain),
    "primaryType": "RFQ",
    "message": asdict(message_to_sign),
}
```